plugins/zoom-to-fit/src/index.js

```diff
@@ -0,0 +1,251 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview UI control for zoom-to-fit.
+ */
+
+import Blockly from 'blockly/core';
+
+/**
+ * Class for zoom to fit control.
+ * @param {!Blockly.WorkspaceSvg} workspace The workspace to sit in.
+ * @implements {Blockly.IPositionable}
+ * @constructor
+ */
+export class ZoomToFitControl {
+  /**
+   * Constructor for zoom-to-fit control.
+   * @param {!Blockly.WorkspaceSvg} workspace The workspace that the plugin will
+   *     be added to.
+   */
+  constructor(workspace) {
+    /**
+     * The workspace.
+     * @type {!Blockly.WorkspaceSvg}
+     * @protected
+     */
+    this.workspace_ = workspace;
+
+    /**
+     * The SVG group containing the zoom controls.
+     * @type {SVGElement}
+     * @private
+     */
+    this.svgGroup_ = null;
+
+    /**
+     * Left coordinate of the zoom controls.
+     * @type {number}
+     * @private
+     */
+    this.left_ = 0;
+
+    /**
+     * Top coordinate of the zoom controls.
+     * @type {number}
+     * @private
+     */
+    this.top_ = 0;
+
+    /**
+     * Width of the zoom controls.
+     * @type {number}
+     * @const
+     * @private
+     */
+    this.WIDTH_ = 32;
+
+    /**
+     * Height of the zoom controls.
+     * @type {number}
+     * @const
+     * @private
+     */
+    this.HEIGHT_ = 32;
+
+    /**
+     * Distance between zoom controls and bottom edge of workspace.
+     * @type {number}
+     * @private
+     */
+    this.MARGIN_BOTTOM_ = 20;
+
+    /**
+     * Distance between zoom controls and right edge of workspace.
+     * @type {number}
+     * @const
+     * @private
+     */
+    this.MARGIN_SIDE_ = 20;
+
+    /**
+     * Whether this has been initialized.
+     * @type {boolean}
+     * @private
+     */
+    this.initialized_ = false;
+  }
+
+  /**
+   * Initializes the zoom reset control.
+   */
+  init() {
+    this.createDom_();
+    this.initialized_ = true;
+  }
+  /**
+   * Disposes of workspace search.
+   * Unlink from all DOM elements and remove all event listeners
+   * to prevent memory leaks.
+   */
+  dispose() {
+    if (this.onZoomOutWrapper_) {
```

Comment:
Will you ever want to dispose of the zoom control other than when the whole workspace is being disposed?

If so, this should remove the svg group from the DOM (although it doesn't need to set `this.svgGroup_` to `null`).

Reply:
done